### ISSUE TITLE
Improve Notification screen layout handling

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#2334]
Actual situation with #2322 
Before & after:
![2018-03-12-081147](https://user-images.githubusercontent.com/5836855/37269950-0c948cbe-25cd-11e8-8dcd-c6d56d6c0b0c.png) ![2018-03-12-081118](https://user-images.githubusercontent.com/5836855/37269951-0cae1e54-25cd-11e8-8edc-5a892b8b2caa.png)

Imho we should handle this similar to how the OS handles longer notifications by showing a shortened (ellipsized) version of the notification and enlarge upon click. cc @nextcloud/designers

**Comment by tobiasKaminsky** [comment]
Good idea with enlarging it. Do we then still want to have some sort of truncation? Or is it then possible to show e.g. 1000chars?

**Comment by AndyScherzinger** [comment]
I guess we are all saying the same thing? :grin: 

We simply would need to agree on the "default size" (as in max lines I guess) and for enlarging I'd say show whatever content the notification contains.

I am looking forward to the @nextcloud/designers feedback + do owe know how the webUI handles this? Since we could always go for "behave like the webUI" in this case.

**Comment by tobiasKaminsky** [comment]
They are doing it the same way as you proposed it, Andy :tada: 
First it is truncated and on click it is showing the complete message.

![image](https://user-images.githubusercontent.com/5836855/37283476-623e36bc-25f8-11e8-86b1-296dc8f8b129.png) ![image](https://user-images.githubusercontent.com/5836855/37283481-67b0f684-25f8-11e8-99b2-70466ea0e687.png)

> We simply would need to agree on the "default size" (as in max lines I guess) 

I would do this somehow relative to the screen size.
If we would say, e.g. 5 lines, it is not the same on tablet vs full hd phone vs low-res phone.

**Comment by AndyScherzinger** [comment]
> They are doing it the same way as you proposed it, Andy
Nice :tada: 

> I would do this somehow relative to the screen size.
If we would say, e.g. 5 lines, it is not the same on tablet vs full hd phone vs low-res phone.

Yes, I'd think that would be a good idea, maybe 4 lines on low res, 5 lines on high res, 7 on anything tablet-like?

**Comment by MorrisJobke** [comment]
I think it should behave like in the web UI: truncated longer text and reveal on click/touch. So :+1: from me on this

**Comment by tobiasKaminsky** [comment]
> Yes, I'd think that would be a good idea, maybe 4 lines on low res, 5 lines on high res, 7 on anything tablet-like?

I thought about dp, or percentage of screen, but I am unsure if this is working. This would then scale automatically instead of hardcoding "arbitrary" lines.

**Comment by jancborchardt** [comment]
We should do it like we do it in the server comments:
If the text content is longer than 4 lines, cut it off visually at 3.5 lines and (if possible) also show a fade to white over it:
![screenshot from 2018-03-19 17-01-11](https://user-images.githubusercontent.com/925062/37607048-46b346cc-2b97-11e8-8329-967f84aa5350.png)
